harvester_e2e_tests/apis/test_hosts.py

```diff
@@ -66,61 +67,80 @@ def test_update_first_node(request, admin_session, harvester_api_endpoints):
     assert resp.status_code == 200, 'Failed to list nodes: %s' % (resp.content)
     host_data = resp.json()
     first_node = host_data['data'][0]
+    original_annotations = copy.deepcopy(first_node['metadata']['annotations'])
     first_node['metadata']['annotations'] = {
         'field.cattle.io/description': 'for-yaml-update',
         'harvesterhci.io/host-custom-name': 'for-yaml-update'
     }
-    resp = utils.poll_for_update_resource(
-        request, admin_session,
-        host_data['data'][0]['links']['update'],
-        first_node,
-        harvester_api_endpoints.get_node % (host_data['data'][0]['id']),
-        use_yaml=True)
-    updated_host_data = resp.json()
-    assert updated_host_data['metadata']['annotations'].get(
-        'field.cattle.io/description') == 'for-yaml-update'
-    assert updated_host_data['metadata']['annotations'].get(
-        'harvesterhci.io/host-custom-name') == 'for-yaml-update'
+    try:
+        resp = utils.poll_for_update_resource(
+            request, admin_session,
+            host_data['data'][0]['links']['update'],
+            first_node,
+            harvester_api_endpoints.get_node % (host_data['data'][0]['id']),
+            use_yaml=True)
+        updated_host_data = resp.json()
+        assert updated_host_data['metadata']['annotations'].get(
+            'field.cattle.io/description') == 'for-yaml-update'
+        assert updated_host_data['metadata']['annotations'].get(
+            'harvesterhci.io/host-custom-name') == 'for-yaml-update'
+    finally:
+        # change it back to original
+        if updated_host_data:
+            updated_host_data['metadata']['annotations'] = original_annotations
+            utils.poll_for_update_resource(
+                request, admin_session,
+                updated_host_data['links']['update'],
+                updated_host_data,
+                harvester_api_endpoints.get_node % (updated_host_data['id']))
 
 
 @pytest.mark.delete_host
 def test_delete_host(request, admin_session, harvester_api_endpoints):
-    cur_endpoint = (request.config.getoption('--endpoint').split(":")[1])[2:]
     resp = admin_session.get(harvester_api_endpoints.list_nodes)
     assert resp.status_code == 200, 'Failed to list nodes: %s' % (resp.content)
     host_data = resp.json()
-    for data in host_data['data']:
-        if data['metadata']['annotations'].get(
-                'etcd.k3s.cattle.io/node-address') != cur_endpoint:
-            host_data_to_delete = data
+    host_data_to_delete = host_data['data'][0]
     # delete the host
     utils.delete_host(request, admin_session, harvester_api_endpoints,
                       host_data_to_delete)
     resp = admin_session.get(harvester_api_endpoints.list_nodes)
     assert resp.status_code == 200, 'Failed to list nodes: %s' % (resp.content)
     host_data = resp.json()
     assert host_data_to_delete not in host_data['data']
+    # TODO(gyee): do we need to make sure the VIP still works?
```

Comment:
we may want to update the configuration entry "harvester_cluster_nodes" in the config.yml after since the subsequent test runs fail the first test test_get_host fails on the assertion

Reply:
good point! I think we may need to run this test separately, last in the pipeline, since it is so destructive.